.github/workflows/pytest-gpu.yml

```diff
@@ -15,85 +11,56 @@ on:
 
-    - name: extract branch name
-      shell: bash
+    - name: Install dependencies
       run: |
-        if [ "${{ github.head_ref }}" == "" ]; then
-          branch="master"
-        else
-          branch="${{ github.head_ref }}"
-        fi
-        echo ::set-output name=branch::$branch
-      id: extract_branch
+        pip3 install --upgrade pip
+        pip3 install -e .[extras]
 
-    - name: find deploy script
+    - name: Test with pytest
       run: |
-        ls '${{ env.OUTPUT_PATH }}/.github/azure'
-
-    - name: deploy VM
-      env:
-        RESOURCE_GROUP: ${{ secrets.CI_GPU_RESOURCE_GROUP }}
-        RESOURCE_GROUP_REGION: uksouth
-        VM_SIZE: Standard_NC6_Promo
-        SERVER_NAME: gpu_ci_vm_${{ steps.extract_branch.outputs.branch }}
-        IMAGE: gpu_ci_vm-image-1226
-      run: >
-        pwsh -command "& '${{ env.OUTPUT_PATH }}\.github\azure\deployVM.ps1'"
-        -servicePrincipal ${{ secrets.SERVICE_PRINCIPAL_APPID }}
-        -servicePrincipalSecret ${{ secrets.SERVICE_PRINCIPAL_SECRET }}
-        -servicePrincipalTenantId ${{ secrets.SERVICE_PRINCIPAL_TENANTID }}
-        -azureSubscriptionName ${{ secrets.AZURE_SUBSCRIPTION_ID }}
-        -resourceGroupName $RESOURCE_GROUP
-        -resourceGroupNameRegion $RESOURCE_GROUP_REGION
-        -size $VM_SIZE
-        -serverName $SERVER_NAME
-        -vmImage $IMAGE
-        -adminLogin ${{ secrets.CI_GPU_VM_ADMIN_LOGIN }}
-        -adminPassword ${{ secrets.CI_GPU_VM_ADMIN_PASSWORD }}
-
-    - name: set host
-      run: echo ::set-output name=action_host::$(az vm show -d -g ${{ secrets.CI_GPU_RESOURCE_GROUP }} -n gpu_ci_vm_${{ steps.extract_branch.outputs.branch }} --query publicIps -o tsv)
-      id: host
+        pytest --cov --cov-config=.coveragerc --cov-report=xml ${{ matrix.test_file }}
+        pytest --cov --cov-config=.coveragerc --cov-report=xml examples/seismic/elastic/elastic_example.py
+        if [ "${{ matrix.name }}" == 'pytest-gpu-omp' ]; then
+            pytest --cov --cov-config=.coveragerc --cov-report=xml examples/seismic/viscoelastic/viscoelastic_example.py
+        fi
 
-    - name: execute tests on VM
-      uses: fifsky/ssh-action@master
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1.0.6
       with:
-        command: |
-          #! /bin/bash
-          cd ~/devito
-          # Ensure no merge conflicts arise
-          git checkout master
-          if [ "${{ steps.extract_branch.outputs.branch }}" == "master" ]; then
-              git pull
-          else
-              git rev-parse --verify --quiet ${{ steps.extract_branch.outputs.branch }}
-              if [ $? -eq 0 ]; then git branch -D ${{ steps.extract_branch.outputs.branch }}; fi
-              git fetch
-              git checkout ${{ steps.extract_branch.outputs.branch }}
-          fi
-          # Set error trap to catch any failing test
-          err=0
-          trap 'err=1' ERR
-          # OpenMP offloading
-          pgaccelinfo
```

Comment:
can we keep this?

Reply:
Yeah, probably should. It's added back in now.